pkg/gitops_test.go

```diff
@@ -530,19 +532,30 @@ func TestCloneGenerateAndPush(t *testing.T) {
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			e := testutils.NewMockExecutor(tt.outputs...)
-			e.Errors = tt.errors
-			err := CloneGenerateAndPush(outputPath, repo, tt.component, e, tt.fs, "main", "/", true)
+
+			outputStack := testutils.NewOutputs(tt.outputs...)
+			executedCmds := []testutils.Execution{}
+			execute = func(baseDir string, cmd CommandType, args ...string) ([]byte, error) {
+				var output []byte
+				var execErr error
+				output, execErr, executedCmds = mockExecute(outputStack, tt.errors, executedCmds, baseDir, cmd, args...)
+				return output, execErr
+			}
+
+			err := CloneGenerateAndPush(outputPath, repo, tt.component, tt.fs, "main", "/", true)
 
 			if tt.wantErrString != "" {
 				testutils.AssertErrorMatch(t, tt.wantErrString, err)
 			} else {
 				testutils.AssertNoError(t, err)
 			}
 
-			assert.Equal(t, tt.want, e.Executed, "command executed should be equal")
+			assert.Equal(t, tt.want, executedCmds, "command executed should be equal")
 		})
 	}
+
+	execute = originalExecute
```

Comment:
What happens if someone forgets to do this in a future test? Will execute be the test execute in non-test code call hierarchy  🤔

Reply:
it is unnecessary to reset after every test as long as the mock function stays the same.  I just did it for the sake of consistency and best practice (in case the mock function does change from test case to test case).  In this particular test case, it was calling a mix of the original and mock execute functions, so the results were getting messed up.   For non-test code, i.e.  it should be the original execute since test functions are not exported